client/src/components/MapBanner.tsx

```diff
@@ -0,0 +1,108 @@
+import React, { useEffect, useState } from "react";
+import { Trans } from "@lingui/macro";
+import Modal from "./Modal";
+import networkDiagram from "../assets/img/network-diagram.png";
+import { logAmplitudeEvent } from "./Amplitude";
+import { CloseButton } from "./CloseButton";
+import classNames from "classnames";
+
+export const PORTFOLIO_SIZE_THRESHOLD = 300;
+
+type MapBannerProps = {
+  message: React.ReactNode | string;
+  buttonText?: React.ReactNode | string;
+  buttonOnClick?: () => void;
+  className?: string;
+  children?: React.ReactNode;
+};
+
+export const MapBanner = ({
+  message,
+  buttonText,
+  buttonOnClick,
+  className,
+  children,
+}: MapBannerProps) => {
+  const [isBanerClosed, setBanerClosed] = useState(false);
+
+  return !isBanerClosed ? (
+    <div className={classNames("MapBanner", className)}>
+      <span className="MapBanner__message">
+        {message} <br />
+        {!!buttonText ? <button onClick={buttonOnClick}>{buttonText}</button> : <></>}
+      </span>
+      <div>
+        <CloseButton onClick={() => setBanerClosed(true)} />
+      </div>
+      {children}
+    </div>
+  ) : (
+    <></>
+  );
+};
+
+type BigPortfolioBannerProps = Omit<
+  MapBannerProps & {
```

Comment:
hmm, wondering if maybe instead of constructing from MapBannerProps, the BigPortfolioBannerProps could just be `sizeOfPortfolio` and `className`, since those appear to be the only ones used?

Reply:
right of course, that didn't make any sense. thanks!